ccf_transaction_processor/Readme.md

```diff
@@ -6,37 +6,123 @@ https://creativecommons.org/licenses/by/4.0/
-TP (except for one additional feature described below). A key difference beween the two ledgers
-is the fact that while the ledger in CCF is encrypted as noted above, the ledger is Sawtooth is stored in plain text.
-Even though the no part of conract state gets stored in the ledger in both CCF & Sawtooth, encrypting the
-ledger as in CCF helps to hide transaction patterns that are otherwise visible in Sawtooth ledger. Detailed documentation about Sawtooth based TP can be found at $PDO_SRC/sawtooth/docs. The schema for JSON payloads used to submit CCF transactions can be found at [${PDO_SRC}/python/pdo/submitter/ccf/docs/](../python/pdo/submitter/ccf/docs/ccf_payload_schema.json). For additional references to documentation about PDO, including transaction processor protocols, see [${PDO_SRC}/README.md](../README.md)
-
-A feature of the CCF based TP that is not supported by Sawtooth based TP is the fact that
-responses to read-transactions include a payload signature, where the signature is generated by the CCF enclave
-serving the read request. The (signing key, verifying key) pair is created within the TP,
-and is globally persisted among all CCF enclaves. The verifying_key can be obtained from CCF using
-the "get_ledger_verifying_key" transaction. This feature may be used by PDO clients to establish offline verifiable proof of transaction commits as desired by the PDO smart contract application. Note that for trust purposes, it is recommended that any entity that uses the above verifying_key gets it directly from the CCF service using the
-"get_ledger_verifying_key" transaction.
-
-We note that the very first invocation of "get_ledger_verifying_key" rpc
-after the service is started is used to create the key pair, which will then be globally committed.
-The system admin who deploys the CCF ledger is expected to invoke the "get_ledger_verifying_key"
-rpc after deployment to generate these keys. Further, the admin must ensure that this key pair is globally committed
-before making the ledger available for PDO usage. This can checked by issuing the rpc again & ensuring that a verifying key gets returned. See https://microsoft.github.io/CCF/users/issue_commands.html for instructions on
-how an rpc can be issued via command line.
+The software is located under $PDO_SRC/transaction_processor/. The TP software is written and tested for CCF tag 0.9.2. Compatability with other CCF versions is not guaranteed.
+
+The instructions below may be followed to build and deploy CCF based PDO TP.
+
+# Get CCF Source Code
+
+CCF tag 0.9.2 is included as a submodule within PDO. Download the submodule using the following commands:
+
+cd $PDO_SRC
+git submodule init
+git submodule update
+
+
+# Install CCF Dependencies
+
+CCF/PDO combo has been tested under a scenario where CCF is deployed in a standalone VM. Further, the CCF/PDO combo
+has been tested only for the CCF virtual enclave mode. The dependencies needed to deploy CCF in an Ubuntu 18.04  VM
```

Comment:
Is it only not tested in HW mode (but should in principle work) or is there any fundamental limitation currently from making it work in HW mode?

Reply:
Thank you for the question. To make CCF work with HW mode, two changes need to be made 1. sign the CCF enclave applicaiton (a few lines to CMakelists.txt) 2. specify OE configuration. We will add these shortly. (I have tested them in Azure, they work)